Exercise 6

pavittr opened the pull request:
Please make sure you are raising this PR against your own repository and not the original. If it says https://github.com/CorndelWithSoftwire in the search bar right now, you are in the wrong place!

Please also don't forget to submit the PR's url to Aptem afterwards (by clicking on the corresponding exercise component, pasting the url in the textbox that appears and then pressing the finish button).

pavittr committed:
Added documentation/context.svg

pavittr committed:
Base Context Diagram

pavittr committed:
Use C4 shapes for Context

pavittr committed:
Added container.svg

pavittr committed:
Update container.svg

pavittr committed:
Added component.svg

pavittr committed:
Update component.svg

pavittr committed:
Update context.svg

pavittr committed:
Update container.svg

hugh-emerson reviewed (approved):
Looks good!

Here are some very small comments, but go ahead and merge this 👍 

Context diagram could maybe emphasise that the Trello system is just a 3rd party tool you're using.

Container diagram:
- You haven't mentioned that the app is running in a container
- You have correctly labelled the web server as "Flask in dev, gunicorn in prod" but the dev detail is probably not necessary
- You haven't mentioned the flask framework anywhere except from the fact that you use Flask's web server in dev
- You said your SPA has "static SPA resources" but that's a lie - the HTML is still dynamically generated and served by the application. 

Component diagram:
- You could maybe mention the jinja2 templates, or the item/view model classes.

pavittr committed:
PR Review: update context.svg

pavittr committed:
PR review: update container.svg

pavittr committed:
PR review: component.svg

pavittr committed:
Context: Add ref to Trello being third party

pavittr committed:
Reference jinja2 in component.svg

pavittr commented:
Thanks for the feedback @hugh-emerson , I've made most fo the changes you suggested:


> Context diagram could maybe emphasise that the Trello system is just a 3rd party tool you're using.
I've added a reference to this being an external third party dependency.

> Container diagram:
> 
> * You haven't mentioned that the app is running in a container
Good point, added

> * You have correctly labelled the web server as "Flask in dev, gunicorn in prod" but the dev detail is probably not necessary

I've removed this now. Its a tricky detail but it does affect the way the app runs, perhaps in future I need two diagrams to cover both scenarios.

> * You haven't mentioned the flask framework anywhere except from the fact that you use Flask's web server in dev
Have added to the description to try and make this clearer. I wonder if this highlights a lack of understanding on my part of what Flask is really doing?

> * You said your SPA has "static SPA resources" but that's a lie - the HTML is still dynamically generated and served by the application.

I'd prefer the term "inaccuracy" as it has fewer connotations 🙂 I haven't implemented this change, because in this case I think I followed a more SPA path (see your previous review comment: https://github.com/pavittr/DevOps-Course-Starter/pull/3#discussion_r666910137). Technically I suppose we do still render the index.html, but it doesn't pass any data and all data is fetched async from the source after page load. I would assuem this is pretty close to the definition of a SPA, but happy to discuss further?

> Component diagram:
> 
> * You could maybe mention the jinja2 templates, or the item/view model classes.
Good point, added a reference to Jinja2 in in the SPA box, and added a new box for the ViewModel (although looking at it, I think removing ViewModel per a previous suggestion you had would be a worthwhile activity).
